### ISSUE TITLE
arcgis-js-api: getProxyRule should accept string

### ISSUE THREAD

**Issue by gobsmack** [DefinitelyTyped/DefinitelyTyped#4336]
esri/urlUtils has a function called getProxyRule. This function should accept a string, which is the url. So, I believe it should be modified like the following:

``` javascript
    /** Returns the proxy rule that matches the given url. */
    getProxyRule(url: string): any;
```

It also seems like there should be a class definition for proxyRule. But, that's not the issue.

**Comment by vvakame** [comment]
@gobsmack please send a pull request. I'll review it.

**Comment by dasa** [comment]
Fixed by #48627

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
